src/ucp/tag/offload.c

```diff
@@ -371,15 +375,86 @@ const ucp_proto_t ucp_tag_offload_proto = {
     .mid_hdr_size     = 0
 };
 
+
 /* Eager sync */
+static size_t ucp_offload_ssend_ack_pack(void *dest, void *arg)
+{
+    ucp_offload_ssend_hdr_t *rep_hdr = dest;
+    ucp_request_t *req               = arg;
+
+    rep_hdr->sender_tag  = req->send.tag_offload.sender_tag;
+    rep_hdr->sender_uuid = req->send.tag_offload.sender_uuid;
+
+    return sizeof(*rep_hdr);
+}
+
+ucs_status_t ucp_offload_progress_sync_ack(uct_pending_req_t *self)
+{
+    ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
+
+    ucs_status_t status = ucp_do_am_bcopy_single(self, UCP_AM_ID_OFFLOAD_SYNC_ACK,
+                                                 ucp_offload_ssend_ack_pack);
+    if (status == UCS_OK) {
+        ucp_request_put(req);
+    }
+    return status;
+}
+
+static ucs_status_t ucp_tag_offload_eager_sync_bcopy(uct_pending_req_t *self)
+{
+    ucp_request_t *req   = ucs_container_of(self, ucp_request_t, send.uct);
+    ucp_worker_t *worker = req->send.ep->worker;
+    ucs_status_t status;
+
+    status = ucp_do_tag_offload_bcopy(self, worker->uuid,
+                                      ucp_tag_offload_pack_eager);
+    if (status == UCS_OK) {
+        ucs_queue_push(&worker->context->tm.sync_reqs,
```

Comment:
bug - need to push to queue before sending first fragment, since ack is sent for first fragment